base/s2i/assemble

```diff
@@ -3,55 +3,66 @@
 # Setup Environment Variables
 LOCAL_SOURCE_DIR="${LOCAL_SOURCE_DIR:-/tmp/src}"
 HYBRIS_HOME="${HYBRIS_HOME:-/opt/hybris}"
+HYBRIS_PLATFORM_FOLDER="${HYBRIS_PLATFORM_FOLDER:-bin/platform}"
 HYBRIS_SOURCE_FOLDER="${HYBRIS_SOURCE_DIR:-hybris}"
+HYBRIS_LOCAL_PROPERTIES_PATH="${HYBRIS_LOCAL_PROPERTIES_PATH-config/local.properties}"
+HYBRIS_SERVER_XML_PATH="${HYBRIS_SERVER_XML_PATH-config/tomcat/conf/server.xml}"
 HYBRIS_CONTENT_DIR="${LOCAL_SOURCE_DIR}/${HYBRIS_SOURCE_FOLDER}"
-ANT_BUILD_TASKS="${ANT_TASKS:-clean all}"
-ANT_DEPLOY_TASKS="${ANT_TASKS:-deploy}"
-ANT_INITIALIZE_TASKS="${ANT_TASKS:-initialize}"
+SOURCE_REPO_LOCAL_PROPERTIES_OVERRIDE="${SOURCE_REPO_LOCAL_PROPERTIES_OVERRIDE}"
+APACHE_JVM_ROUTE_NAME="${APACHE_JVM_ROUTE_NAME}"
+HYBRIS_ANT_TASK_NAMES="${HYBRIS_ANT_TASK_NAMES}"
 
 if [ ! -d "${HYBRIS_CONTENT_DIR}" ]; then
     echo "Could Not Locate Hybris Directory in Source!"
     exit 1
 fi
 
 # Copying Source to Hybris Server
-echo "---> Copying Hybris application source ..."
-cp -r ${HYBRIS_CONTENT_DIR}/* "${HYBRIS_HOME}"
+echo "---> Copying Hybris application source cp -r ${HYBRIS_CONTENT_DIR}/* ${HYBRIS_HOME}"
+cp -r ${HYBRIS_CONTENT_DIR}/* ${HYBRIS_HOME}
 
-# Removing Hybris Source
-rm -rf "${HYBRIS_CONTENT_DIR}"
+# Override Hybris Server local.properties 
+if [ ! -z "${SOURCE_REPO_LOCAL_PROPERTIES_OVERRIDE}" ]; then
+    echo "---> Hybris config  ${HYBRIS_HOME}/${HYBRIS_LOCAL_PROPERTIES_PATH} is replaced by ${SOURCE_REPO_LOCAL_PROPERTIES_OVERRIDE}"
+    curl -k -o "${HYBRIS_HOME}/${HYBRIS_LOCAL_PROPERTIES_PATH}" "${SOURCE_REPO_LOCAL_PROPERTIES_OVERRIDE}"
+else
+    echo "---> Hybris config ${HYBRIS_HOME}/${HYBRIS_LOCAL_PROPERTIES_PATH} present in the source repo will be used as is"
+fi
+
+# Replace Apache Tomcat server.xml for a jvmRoute value
+if [ ! -z "${APACHE_JVM_ROUTE_NAME}" ]; then
+    echo "---> Apache config  ${HYBRIS_HOME}/${HYBRIS_SERVER_XML_PATH} modified for jvmRoute with a value of 'ChangeMe' to '${APACHE_JVM_ROUTE_NAME}'"
+    sed -i s/ChangeMe/"${APACHE_JVM_ROUTE_NAME}"/g ${HYBRIS_HOME}/${HYBRIS_SERVER_XML_PATH}
+else
+    echo "---> Apache config ${HYBRIS_HOME}/${HYBRIS_SERVER_XML_PATH} present in the source repo will be used as is"
+fi
 
 
 # Build Hybris Server
 echo "---> Building SAP Commerce ..."
-cd "${HYBRIS_HOME}/bin/platform/"
+cd ${HYBRIS_HOME}/${HYBRIS_PLATFORM_FOLDER}
 . ./setantenv.sh
-ant ${ANT_BUILD_TASKS}
 
-if [ $? -ne 0 ]; then
-    echo
-    echo "SAP Commerce Build Failed!"
-    exit 1
+# Execute multiple ANT tasks for Hybris one at a time
+if [ ! -z "${HYBRIS_ANT_TASK_NAMES}" ]; then
+    echo "---> All Hybris ANT tasks execution started for ${HYBRIS_ANT_TASK_NAMES}"
+    IFS=","
+    for v in ${HYBRIS_ANT_TASK_NAMES}
+    do
+        echo "---> Executing ant $v"
+        ant $v
```

Comment:
The build should fail immediately if any `ant` task fails.

Reply:
I am making a change here to break on ANT task failure. I agree with you here